### ISSUE TITLE
Add EVA-02 / EVA-02-CLIP vision encoder to Transformers

### ISSUE THREAD

**Issue by Aaraviitkgp** [huggingface/transformers#42475]
### Model description

EVA-02 and EVA-02-CLIP are state-of-the-art vision encoders released by the BAIDU / IDEA Research team.
They achieve extremely strong performance across image classification, retrieval, and multimodal benchmarks, and serve as the visual backbone in several high-performing VLMs (e.g., LLaVA-Next, Qwen-VL).

### Open source status

- [x] The model implementation is available
- [x] The model weights are available

### Provide useful links for the implementation

Code (MIT License):
	•	EVA GitHub: https://github.com/baaivision/EVA

Model weights:
	•	EVA-02-CLIP models on HuggingFace Hub:
	•	https://huggingface.co/QuanSun/EVA-CLIP

Papers:
	•	EVA-02: A Visual Representation Model for the Era of Foundation Models
                 https://arxiv.org/abs/2308.02201

Authors:
	•	BAAI / IDEA Research team

cc @molbap @yonigozlan @zucchini-nlp

**Comment by Rocketknight1** [comment]
Hmn, if they're the vision backbone for existing models, that means their code is in Transformers already!

**Comment by Aaraviitkgp** [comment]
@Rocketknight1 It is partially implemented in Qwen2VL as Qwen2VLVisionTransformer but no standalone implementation is there.

**Comment by Rocketknight1** [comment]
Hmmn - cc @molbap @yonigozlan @zucchini-nlp in that case, do we want support for the standalone vision encoder?

**Comment by zucchini-nlp** [comment]
Seems to be supported in timm already, I can see Eva and Eva-2 in https://github.com/huggingface/pytorch-image-models?tab=readme-ov-file#models 👀

**Comment by Aaraviitkgp** [comment]
@zucchini-nlp timm supports eva and eva-2 but having a native Eva02Config and Eva02Model would help, native support would allow nesting vision configuration directly inside vlm configuration which will allow the entire multimodal to be initialized,trained and saved as single unit which will allow users to use feature like gradient checkpointing and device_map="auto" usage which is complicated via timm.

**Comment by zucchini-nlp** [comment]
Transformers already supports all the models from timm as standalone or as part of VLMs thanks to [`TimmWrapper`](https://huggingface.co/docs/transformers/en/model_doc/timm_wrapper). For ex gemma3 models' vision encoder is not implemented in transformers, we import and re-use from timm
